code/game/machinery/clonescanner.dm

```diff
@@ -0,0 +1,231 @@
+	update_icon()
+	RefreshParts()
+
+/obj/machinery/clonescanner/RefreshParts()
+	for(var/obj/item/stock_parts/SP in component_parts)
+		scanning_tier += SP.rating
+
+
+/obj/machinery/clonescanner/MouseDrop_T(atom/movable/O, mob/user)
+	if(!(ishuman(user) || issilicon(user)) || user.incapacitated())
+		return
+	if(!ishuman(O))
+		return
+	var/mob/living/carbon/human/H = O
+	if(H.stat != DEAD)
+		to_chat(user, "<span class='warning'>You don't think it'd be wise to scan a living being.</span>")
+		return TRUE
+	if(occupant)
+		to_chat(user, "<span class='warning'>[src] is already occupied!</span>")
+		return TRUE
+
+	to_chat(user, "<span class='notice'>You put [H] into the cloning scanner.</span>")
+	insert_mob(H)
+	return TRUE
+
+/obj/machinery/clonescanner/AltClick(mob/user)
+	if(!occupant)
+		return
+	if(issilicon(user))
+		remove_mob(occupant)
+		return
+	if(!Adjacent(user) || !ishuman(user) || HAS_TRAIT(user, TRAIT_HANDS_BLOCKED))
+		return
+	remove_mob(occupant)
+
+/obj/machinery/clonescanner/relaymove(mob/user)
+	if(user.stat)
+		return
+	remove_mob(user)
+
+/obj/machinery/clonescanner/proc/try_scan(mob/living/carbon/human/scanned)
+	if(!scanned)
+		return
+
+	occupant.notify_ghost_cloning()
+
+	has_scanned = TRUE
+
+	if(!scanned.dna || HAS_TRAIT(scanned, TRAIT_GENELESS))
+		return SCANNER_MISC
+	if(HAS_TRAIT(scanned, TRAIT_BADDNA) && scanning_tier < 4)
+		return SCANNER_ABSORBED
+	if(HAS_TRAIT(scanned, TRAIT_HUSK) && scanning_tier < 4)
+		return SCANNER_HUSKED
+	if(NO_CLONESCAN in scanned.dna.species.species_traits)
+		return SCANNER_UNCLONEABLE_SPECIES
+	if(!scanned.ckey || !scanned.client || ischangeling(scanned))
+		return SCANNER_NO_SOUL
+	if(scanned.suiciding || !scanned.get_int_organ(/obj/item/organ/internal/brain))
+		return SCANNER_BRAIN_ISSUE
+
+	return scan(scanned)
+
+/obj/machinery/clonescanner/proc/scan(mob/living/carbon/human/scanned)
+	var/datum/cloning_data/scan_result = new
+
+	scan_result.name = scanned.dna.real_name
+	scan_result.mindUID = scanned.mind.UID()
+	scan_result.genetic_info = scanned.dna.Clone()
+
+	for(var/limb in scanned.dna.species.has_limbs)
+		if(scanned.bodyparts_by_name[limb])
+			var/obj/item/organ/external/active_limb = scanned.bodyparts_by_name[limb]
+			scan_result.limbs[limb] = list(active_limb.brute_dam,
+											active_limb.burn_dam,
+											active_limb.status,
+											FALSE,
+											active_limb.name,
+											active_limb.max_damage)
+		else
+			scan_result.limbs[limb] = list(0, 0, 0, TRUE, scanned.dna.species.has_limbs[limb]["descriptor"], 0) //no damage if it's missing!
+
+	for(var/organ in scanned.dna.species.has_organ)
+		var/obj/item/organ/internal/active_organ = scanned.get_int_organ(scanned.dna.species.has_organ[organ]) //this is icky
+		if(istype(active_organ))
+			scan_result.organs[organ] = list(active_organ.damage,
+											active_organ.status,
+											FALSE,
+											active_organ.name,
+											active_organ.max_damage,
+											active_organ.organ_tag)
```

Comment:
what is going on with these lists and the FALSE's on the far right?

Reply:
This is constructing a `cloning_data` datum from the organ's current stats, the `FALSE` is the 'missing' field
![image](https://github.com/ParadiseSS13/Paradise/assets/47290811/ca9059ca-f801-4ed8-bdc4-07c960bde6cf)

Reply:
Apologies, it looks like this in github 
![image](https://github.com/ParadiseSS13/Paradise/assets/12197162/99460bcc-d5c0-411f-a22a-0430d05dc339)

Reply:
which is a little confusing